src/Frontend/Core/Engine/Url.php

```diff
@@ -364,9 +364,12 @@ private function processQueryString()
 
             // remove language
             if ($hasMultiLanguages) {
-                $URL = trim(str_replace('/' . $language, '', $URL), '/');
+                $URL = str_replace('/' . $language, '', $URL);
             }
 
+            // remove the first slash
+            $URL = trim($URL, '/');
```

Comment:
ok. I'll test this one before merging it, because I can't predict the consequences of this change without testing. Thanks for the PR!